src/dali_executor/dali_pipeline.h

```diff
@@ -226,14 +228,14 @@ class DaliPipeline {
   void CreatePipeline() {
     daliCreatePipeline(&handle_, serialized_pipeline_.c_str(), serialized_pipeline_.length(),
                        max_batch_size_, num_threads_, device_id_, 0, 1, 0, 0, 0);
-    assert(handle_.pipe != nullptr && handle_.ws != nullptr);
+    dali_pipeline_created_ = true;
```

Comment:
This isn't necessary.

Reply:
Unfortunately, this is necessary. Otherwise this code (which actually happens during initialization) won't work:
```c++
DaliPipeline p1;
DaliPipeline p2;
p2.CreatePipeline();
p2 = std::move(p1);
```

Reply:
But why wouldn't it work? Moving p2 to p1 would just swap the handles and now the `p2 ` looks as if it wasn't ever initialized (it will have a null handle).